File: Data Structures & Algorithms/count-paths/submission-0.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[][] ans = new int[m][n];
        ans[0][0] = 1;
        for(int i = 0; i<m; i++)
        {
            for(int j = 0; j<n; j++)
            {
                if(i-1<0 && j-1>=0)
                {
                    ans[i][j] = ans[i][j-1];
                }
                if(j-1<0 && i-1>=0)
                {
                    ans[i][j] = ans[i-1][j];
                }   
                else if(j-1>=0 && i-1>=0)
                {
                    ans[i][j] = ans[i-1][j] + ans[i][j-1];
                }
            }
        }
        return ans[m-1][n-1];
    }
}

File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int> num;
    vector<bool> visited;
    vector<vector<int>> permute(vector<int>& nums) {
        num = nums;
        visited = vector<bool>(nums.size(), false);
        vector<int> temp;
        recurse(temp);
        return ans;
    }

    void recurse(vector<int> temp)
    {
        if(temp.size()==num.size())
        {
            ans.push_back(temp);
            return;
        }
        for(int i = 0; i<num.size(); i++)
        {  
            if(!visited[i])
            {
                visited[i] = true;
                temp.push_back(num[i]);
                recurse(temp);
                visited[i] = false;
                temp.pop_back();
            }
        }
    }
};